HW2/P3/mandelbrot.pyx

```diff
@@ -5,73 +5,168 @@ import numpy
 cimport AVX
 from cython.parallel import prange
 
-
 cdef np.float64_t magnitude_squared(np.complex64_t z) nogil:
-    return z.real * z.real + z.imag * z.imag
+	return z.real * z.real + z.imag * z.imag
+
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+
+
+cpdef mandelbrot_AVX(np.complex64_t [:, :] in_coords,
+				np.uint32_t [:, :] out_counts,
+				int n_threads,
+				int max_iterations=511):
+	cdef:
+		int i, j, iter, k
+		np.float32_t [:, :] in_coords_real, in_coords_imag
+		AVX.float8 c_real, c_imag, iter_AVX, z_mod
+		AVX.float8 mask, z_real, z_imag, z_real_temp
+
+
+	in_coords_real = np.real(in_coords)
+	in_coords_imag = np.imag(in_coords)
+
+
+	# parallize the rows with prange
+	for i in prange(in_coords.shape[0], nogil = True, schedule = 'static', chunksize =1,
+	 				num_threads = n_threads):
+
+		# we now go by steps of 8 within the loop because each AVX takes care of 8 values
+		for j in range(0,in_coords.shape[1],8):
+			# define c_real (in the AVX 7 to 0 format)
+			c_real = AVX.make_float8(in_coords_real[i, j+7], in_coords_real[i, j+6], in_coords_real[i, j+5],
```

Comment:
As a style thing, it might be good to put each of the arguments on it's own line, and do some indentation (this is possible in python if the statements are enclosed in parentheses.